tests/unit/h/views/account_signup_test.py

```diff
@@ -41,7 +66,7 @@ def test_post_creates_user_from_form_data(
             username="bob",
             email="bob@example.com",
             password="s3crets",  # noqa: S106
-            privacy_accepted=datetime.datetime.utcnow.return_value,
+            privacy_accepted=datetime.datetime.now(datetime.UTC),
```

Comment:
Does `datetime.datetime.now` return a fixed value in the context of tests or could this change in between the `post` call and the `assert_called_with` call @seanh?

Reply:
@robertknight FYI it's very unlikely that I will notice if you `@mention` me on GitHub, my GitHub notifications are full of so much noise, I don't check them.

`datetime.datetime.now()` does not normally return a fixed value in tests. The correct solution is to use [freezegun](https://github.com/spulec/freezegun) (which is already a dependency in this repo) to make it do so on a per-test basis.

The reason why this test is passing is that the test module [patches `datetime` with a mock](https://github.com/hypothesis/h/blob/3686c9d103b4943b49576978def2924997195415/tests/unit/h/views/account_signup_test.py#L120-L122), which is also fine although the patch fixture [should have `autouse=True`](https://hypothes-is.slack.com/archives/C1MA4E9B9/p1718193955021929).